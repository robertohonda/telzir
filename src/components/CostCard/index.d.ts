import React from 'react'
import { StandardProps } from '@material-ui/core';

export interface CostCardProps extends StandardProps<{}, CostCardClassKey> {
  logoPath: string
  cost: string
}

export type CostCardClassKey =
  | 'title'

declare const CostCard: React.ComponentType<CostCardProps>

export default CostCard