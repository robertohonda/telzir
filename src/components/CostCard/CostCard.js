import React from 'react'
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core/styles'
import Card from '@material-ui/core/Card'
import CardHeader from '@material-ui/core/CardHeader'
import Typography from '@material-ui/core/Typography'
import CardContent from '@material-ui/core/CardContent'
import CircularProgress from '@material-ui/core/CircularProgress'

const useStyles = makeStyles((theme) => ({
  title: {
    color: '#6E6E6E',
  },
}))
const CostCard = (props) => {
  const classes = useStyles(props)
  const { title, cost, loading } = props
  return (
    <Card>
      <CardHeader
        title={
          <Typography variant="h5" className={classes.title}>
            {title}
          </Typography>
        }
      />
      <CardContent>
        {loading ? (
          <div style={{ textAlign: 'center' }}>
            <CircularProgress size={44} className={classes.progress} />
          </div>
        ) : (
          <Typography color="primary" variant="h3" align="center">
            {cost}
          </Typography>
        )}
      </CardContent>
    </Card>
  )
}

CostCard.propTypes = {
  title: PropTypes.string.isRequired,
  cost: PropTypes.string.isRequired,
  loading: PropTypes.bool,
}

export default CostCard
