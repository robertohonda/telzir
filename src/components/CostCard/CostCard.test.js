import React from 'react'
import { shallow } from 'enzyme'
import CostCard from './CostCard'

describe('CostCard', () => {
  it('should render correctly', () => {
    const component = shallow(<CostCard title="Title" cost="20.00" />)

    expect(component).toMatchSnapshot()
  })
})
