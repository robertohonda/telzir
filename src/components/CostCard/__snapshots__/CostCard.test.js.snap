// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CostCard should render correctly 1`] = `
<WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(CardHeader))
    title={
      <ForwardRef(WithStyles)
        className="makeStyles-title-1"
        variant="h5"
      >
        Title
      </ForwardRef(WithStyles)>
    }
  />
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      align="center"
      color="primary"
      variant="h3"
    >
      20.00
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
